quickstack/manifests/sahara.pp

```diff
@@ -68,6 +69,135 @@
+      path => "${base_dir}nodegroup_templates/panel.py",
+      line => "    permissions = (('openstack.roles.sahara_user','openstack.roles.sahara_admin'),)",
+      after => "                    'openstack.services.data_processing'\u0029,\u0029"
+    }
+    file_line { 'wizard':
+      notify  => Service['httpd'], # only restarts if change
+      path => "${base_dir}wizard/panel.py",
+      line => "    permissions = (('openstack.roles.sahara_user','openstack.roles.sahara_admin'),)",
+      after => "                    'openstack.services.data_processing'\u0029,\u0029"
+    }
+    file_line { 'data_image_registry':
+      notify  => Service['httpd'], # only restarts if change
+      path => "${base_dir}data_image_registry/panel.py",
+      line => "    permissions = (('openstack.roles.sahara_admin',),)",
+      after => "                    'openstack.services.data_processing'\u0029,\u0029"
+    }
+  }
+  
+  $m = '/usr/lib/python2.7/site-packages/sahara/plugins/mapr/versions'
+  $mapr_dirs = [ "${m}/mapr_spark", "${m}/v3_1_1", "${m}/v4_0_1_mrv1", "${m}/v4_0_1_mrv2",
+                 "${m}/v4_0_2_mrv1", "${m}/v4_0_2_mrv2", "${m}/v5_0_0_mrv1" ]
+
+  file { $mapr_dirs :
+    notify  => Service['openstack-sahara-all'], # only restarts Sahara if a file changes
+    ensure  => absent,
+    recurse => true,
+    purge   => true,
+    force   => true,
+  }
+
+  file { 'old_vanilla':
+    notify  => Service['openstack-sahara-all'], # only restarts Sahara if a file changes
+    ensure  => absent,
+    path    => '/usr/lib/python2.7/site-packages/sahara/plugins/vanilla/v2_6_0',
+    recurse => true,
+    purge   => true,
+    force   => true,
+  }
+  
+  file_line { 'spark_cleanup':
+    notify  => Service['openstack-sahara-all'], # only restarts Sahara if a file changes
+    path    => '/usr/lib/python2.7/site-packages/sahara/plugins/spark/plugin.py',
+    line    => '        return ["1.3.1"]',
+    after   => "    def get_versions\u0028self\u0029:"
+  }
+
+  file_line { 'ambari_cleanup':
+    notify  => Service['openstack-sahara-all'], # only restarts Sahara if a file changes
+    path    => '/usr/lib/python2.7/site-packages/sahara/plugins/ambari/plugin.py',
+    line    => '        return ["2.2"]',
+    after   => "    def get_versions\u0028self\u0029:"
+  }
+
+  file_line { 'mapr_hue_fix':
+    notify  => Service['openstack-sahara-all'], # only restarts Sahara if a file changes
+    path    => '/usr/lib/python2.7/site-packages/sahara/plugins/mapr/services/hue/hue.py',
+    line    => 'HueV381 = HueV370'
+  }
+
+  file_line { 'sencha_mapr':
+    notify  => Service['openstack-sahara-all'], # only restarts Sahara if a file changes
+    path    => '/usr/lib/python2.7/site-packages/sahara/plugins/mapr/services/oozie/oozie.py',
+    line    => "        extjs_url = 'http://sahara-files.mirantis.com/ext-2.2.zip'",
+    after   => "        extjs_url = 'http://dev.sencha.com/deploy/ext-2.2.zip'"
+  }
+
+  file_line { 'sencha_cdh':
```

Comment:
I think using file_line is good enough, especially since almost of these changes only are just for Liberty... when we upgrade to Mitaka they are no longer necessary.